internal/resources/iampolicy/workspace_scope.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright © 2022 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+// nolint: dupl
+package iampolicy
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper"
+	workspacemodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/workspace"
+)
+
+var workspaceFullname = &schema.Schema{
+	Type:        schema.TypeList,
+	Description: "The schema for workspace iam policy full name",
+	Optional:    true,
+	MaxItems:    1,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			workspaceNameKey: {
+				Type:        schema.TypeString,
+				Description: "Name of the workspace",
+				Required:    true,
+				ForceNew:    true,
+			},
+		},
+	},
+}
+
+func constructWorkspaceFullname(data []interface{}) (fullname *workspacemodel.VmwareTanzuManageV1alpha1WorkspaceFullName) {
+	if len(data) == 0 || data[0] == nil {
+		return fullname
+	}
+
+	fullNameData, _ := data[0].(map[string]interface{})
+
+	fullname = &workspacemodel.VmwareTanzuManageV1alpha1WorkspaceFullName{}
+
+	if v, ok := fullNameData[workspaceNameKey]; ok {
+		helper.SetPrimitiveValue(v, &fullname.Name, workspaceNameKey)
+	}
+
+	return fullname
+}
+
+func flattenWorkspaceFullname(fullname *workspacemodel.VmwareTanzuManageV1alpha1WorkspaceFullName) (data []interface{}) {
+	if fullname == nil {
+		return data
+	}
+
+	flattenFullname := make(map[string]interface{})
+
+	flattenFullname[workspaceNameKey] = fullname.Name
+
+	return []interface{}{flattenFullname}
+}
```

Comment:
Please define these in the appropriate resource package and reuse them to avoid duplication and redundancy in the codebase, as the models are standard.